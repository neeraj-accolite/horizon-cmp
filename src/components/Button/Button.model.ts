import { ButtonProps as CustomButtonProps } from '@nextui-org/button';

export interface ButtonProps extends CustomButtonProps {
  /**
   * The class applied to the parent element.
   */
  rootClass?: string;
  /**
   * Label for the button
   */
  label: string;
  /**
   * Label Color
   */
  labelColor?: string;
  /**
   * Font weight of the button
   */
  fontWeight?: number;
  /**
   * Border Color for border
   */
  borderColor?: string;
  /**
   * OnClick event of the buttons
   */
  onClick?: () => void;
}
