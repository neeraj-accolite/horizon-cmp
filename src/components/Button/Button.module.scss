.button {
  @apply rounded px-4 py-2 text-sm font-bold;
}

.primary {
  @apply bg-red-500 text-white hover:bg-blue-600;
}

.secondary {
  @apply bg-gray-500 text-white hover:bg-gray-600;
}
