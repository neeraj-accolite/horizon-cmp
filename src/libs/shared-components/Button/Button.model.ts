import { ButtonProps as CustomButtonProps } from '@nextui-org/button';

export interface ButtonProps extends CustomButtonProps {
  /**
   * The class applied to the parent element.
   */
  className?: string;
  /**
   * url of the button
   */
  url: string;
  /**
   * The colors for the button
   */
  color: 'default' | 'primary' | 'secondary' | 'success' | 'warning' | 'danger';
  /**
   * isDisable button to enable and disable the button
   */
  isDisabled?: boolean;
  /**
   * A function to be called when the button is clicked.
   * This handles the click event for the button.
   */
  onClick?: () => void;
}
