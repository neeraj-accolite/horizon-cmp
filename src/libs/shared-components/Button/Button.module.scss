.button {
  @apply h-10 gap-2 px-4 font-bold leading-6;
}
