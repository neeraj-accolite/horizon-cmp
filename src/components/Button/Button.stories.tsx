import Button from "./Button";
import { Meta, StoryObj } from "@storybook/react";

export default {
  title: "Components/Button",
  component: Button,
  parameters: { layout: "centered" },
} as Meta<typeof Button>;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    label: "Primary Button",
    color: "primary",
    onClick: () => console.log("Primary button is clicked"),
    disabled: false,
  },
};

export const Secondary: Story = {
  args: {
    label: "Secondary Button",
    color: "secondary",
    onClick: () => console.log("Secondary button is clicked"),
    disabled: false,
  },
};
