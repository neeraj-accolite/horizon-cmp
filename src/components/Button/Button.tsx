import React from 'react';
import { ButtonProps } from './Button.model';
import styles from './Button.module.scss';
import { Button as ButtonComp } from '@nextui-org/button';

const Button = ({
  label,
  onClick,
  color = 'primary',
  ...rest
}: ButtonProps) => {
  return (
    <ButtonComp
      {...rest}
      disableAnimation={true}
      className={`${styles.button} ${styles[color]}`}
      onPress={onClick}
      aria-label={label}
    >
      {label}
    </ButtonComp>
  );
};

export default Button;
