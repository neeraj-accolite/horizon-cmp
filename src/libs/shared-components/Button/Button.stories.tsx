import Button from './Button';
import { Meta, StoryObj } from '@storybook/react';
import { Star } from '../Icons';
export default {
  title: 'Components/Button',
  component: Button,
  parameters: { layout: 'centered' },
} as Meta<typeof Button>;

type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    children: 'Primary Button',
    onClick: () => console.log('Primary button is clicked'),
    isDisabled: false,
    size: 'md',
    radius: 'sm',
    color: 'primary',
  },
};

export const Secondary: Story = {
  args: {
    children: 'Secondary Button',
    onClick: () => console.log('Secondary button is clicked'),
    isDisabled: false,
    size: 'md',
    radius: 'sm',
    className: 'w-full',
    color: 'secondary',
  },
};

export const Outline: Story = {
  args: {
    children: 'Outline Button',
    onClick: () => console.log('Outline button is clicked'),
    isDisabled: false,
    size: 'sm',
    radius: 'sm',
    color: 'primary',
    variant: 'bordered',
  },
};

export const Danger: Story = {
  args: {
    children: 'Danger Button',
    color: 'danger',
    onClick: () => console.log('Danger button is clicked'),
    isDisabled: false,
    size: 'md',
    radius: 'sm',
    className: 'w-full',
  },
};

export const Success: Story = {
  args: {
    children: 'Success Button',
    color: 'success',
    onClick: () => console.log('Success button is clicked'),
    isDisabled: false,
    size: 'md',
    radius: 'sm',
    className: 'w-full',
  },
};

export const Warning: Story = {
  args: {
    children: 'Warning Button',
    color: 'warning',
    onClick: () => console.log('Warning button is clicked'),
    isDisabled: false,
    size: 'md',
    radius: 'sm',
    className: 'w-full',
  },
};

export const SmallOutline: Story = {
  args: {
    children: 'Small Outline Button',
    onClick: () => console.log('Small Outline button is clicked'),
    isDisabled: false,
    size: 'sm',
    radius: 'sm',
    className: 'border-blue-500',
    color: 'primary',
    variant: 'bordered',
  },
};

export const LargeOutline: Story = {
  args: {
    children: 'Large Outline Button',
    onClick: () => console.log('Large Outline button is clicked'),
    isDisabled: false,
    size: 'lg',
    radius: 'lg',
    className: 'border-blue-500',
    color: 'primary',
    variant: 'bordered',
  },
};

export const CustomWidth: Story = {
  args: {
    children: 'Custom Width Button',
    onClick: () => console.log('Custom Width button is clicked'),
    isDisabled: false,
    className: 'w-[300px]',
    color: 'primary',
  },
};

export const Disabled: Story = {
  args: {
    children: 'Disabled Button',
    onClick: () => console.log('Disabled button is clicked'),
    isDisabled: true,
    size: 'md',
    radius: 'sm',
    className: 'w-full',
    color: 'primary',
  },
};

export const Loading: Story = {
  args: {
    children: 'Loading Button',
    color: 'primary',
    isLoading: true,
    onClick: () => console.log('Loading button is clicked'),
    isDisabled: false,
    size: 'md',
    radius: 'sm',
    className: 'w-full',
  },
};

export const WithStartEndContent: Story = {
  args: {
    children: 'Button with Content',
    color: 'primary',
    startContent: <span>ðŸŒŸ</span>,
    endContent: <span>ðŸš€</span>,
    onClick: () => console.log('Button with content is clicked'),
    isDisabled: false,
    size: 'md',
    radius: 'sm',
    className: 'w-full',
  },
};

export const StarIconButton: Story = {
  args: {
    children: <Star size={20} />,
    isIconOnly: true,
    color: 'danger',
    size: 'lg',
    onClick: () => console.log('Star button clicked'),
  },
};
