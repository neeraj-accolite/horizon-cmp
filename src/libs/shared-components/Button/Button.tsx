import React from 'react';
import { Button as NextUIButton } from '@nextui-org/button';
import { ButtonProps } from './Button.model';
import styles from './Button.module.scss';

const Button: React.FC<ButtonProps> = ({
  className = '',
  children,
  url,
  isDisabled = false,
  radius = 'sm',
  color,
  variant,
  onClick,
  ...rest
}) => {
  return (
    <NextUIButton
      as="a"
      href={isDisabled ? undefined : url}
      variant={variant}
      isDisabled={isDisabled}
      className={`${styles.button} ${variant === 'bordered' ? styles.outline : styles[color]} ${className}`}
      onPress={onClick}
      radius={radius}
      color={color}
      {...rest}
    >
      {children}
    </NextUIButton>
  );
};

export default Button;
