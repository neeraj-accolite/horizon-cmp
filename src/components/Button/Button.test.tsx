import React from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';
import Button from './Button';

describe('UI Component - Button', () => {
  test('render the button component with default props ', async () => {
    const container = render(<Button label="Default Button" />);
    const button = await container.findByText('Default Button');
    expect(button).toBeInTheDocument();
  });

  test('onClick method is called when the button is clicked', async () => {
    const handleClick = jest.fn();
    const container = render(<Button label="Click Me" onClick={handleClick} />);
    const button = await container.findByText('Click Me');
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('onClick method should not be called when disabled', async () => {
    const handleClick = jest.fn();
    const container = render(
      <Button label="Click Me" onClick={handleClick} disabled={true} />,
    );
    const button = await container.findByText('Click Me');
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(0);
  });

  test('render the button component unchanged with default parameters', async () => {
    const container = render(
      <Button label="Primary Button" variant="primary" labelColor="#ff0000" />,
    );
    expect(container).toMatchSnapshot();
  });
});
