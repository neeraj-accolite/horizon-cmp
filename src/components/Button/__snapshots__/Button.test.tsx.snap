// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Component - Button render the button component unchanged with default parameters 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="Primary Button"
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover button primary"
        type="button"
      >
        Primary Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="Primary Button"
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] !transition-none data-[pressed=true]:scale-100 bg-default text-default-foreground data-[hover=true]:opacity-hover button primary"
      type="button"
    >
      Primary Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;